### ISSUE TITLE
Specification Decision Needed: Translation of Sparkplug Definitions to Edge AI Systems

### ISSUE THREAD

**Issue by caradoxical** [modzy/edge-ai-standards-mqtt#4]
Within Sparkplug, all messages must adhere to the following topic structure:

`namespace/group_id/message_type/edge_node_id/[device_id]`

Within this structure, we must decide how to recommend using the `edge_node_id` and `device_id` elements. Our specification currently recommends using these elements as follows:

* **edge_node_id**: Edge nodes are defined as any client that manages an MQTT session. Based on this definition it is then recommended to use this element to indicate the identity of the computing hardware that is actively running models and connected to an MQTT session. Examples could include:
  * `ip_camera_1` An IP camera with an embedded GPU that is actively running one or more models.
  * `edge_server_1` A dedicated, on-site hosting server that is actively running one or more models.
  * `machinery_1` A piece of machinery with enough on-board processing power to run an AI/ML model locally. This may be the case for embedded models running on a microcontroller.
* **device_id**: While often used to to indicate a physical or logical device, such as a PLC, this field can also be used to represent a logical grouping of data points. Using this latter definition, device_id should be set to the name of the model that is generating inferences. A human-readable model name with the model's version number appended (i.e. `Machine Failure Prediction:0.0.1`) is a very readable way to structure this value, however care should be taken to choose a value for the model name that is stable. This may mean that using a unique identifier is actually the best choice (i.e. `brzrip6cxk:0.0.1`) so that a change in model name doesn't impact downstream applications.

In summary: 

`edge_node_id` -> The AI processing hardware (i.e. Rasperry Pi, Jetson Nano, IP Camera with an embedded GPU, etc.)
`device_id` -> Each AI model running on or connected to an `edge_node_id`

**Comment by caradoxical** [comment]
12/8/23 discussion:

- Total agreement about  interpretation of Sparkplug top level topic names

**Comment by caradoxical** [comment]
Decision was made to accept interpretation initially proposed by Kudzai, for adapting Sparkplug namespace elements to edge AI models was made on 12/8/23
